packages/utils/src/formatDate.ts

```diff
@@ -0,0 +1,72 @@
+import { differenceInDays } from "date-fns"
+import { formatRemainingTime } from "./formatRemainingTime"
+
+const DEFAULT_DAYS_RELATIVE_THRESHOLD = 3
+const DEFAULT_LOCALE = "nb-NO"
+
+const DEFAULT_INTL_DATE_FORMAT_OPTIONS = {
+  day: "2-digit",
+  month: "2-digit",
+  year: "numeric",
+} satisfies Intl.DateTimeFormatOptions
+
+const DEFAULT_INTL_WEEKDAY_FORMAT_OPTIONS = {
+  weekday: "long",
+} satisfies Intl.DateTimeFormatOptions
+
+const DEFAULT_INTL_TIME_FORMAT_OPTIONS = {
+  hour: "2-digit",
+  minute: "2-digit",
+} satisfies Intl.DateTimeFormatOptions
+
+export const IntlFormats = {
+  Date: DEFAULT_INTL_DATE_FORMAT_OPTIONS,
+  Time: DEFAULT_INTL_TIME_FORMAT_OPTIONS,
+  Weekday: DEFAULT_INTL_WEEKDAY_FORMAT_OPTIONS,
+  DateTime: { ...DEFAULT_INTL_DATE_FORMAT_OPTIONS, ...DEFAULT_INTL_TIME_FORMAT_OPTIONS },
+  DateWeekday: { ...DEFAULT_INTL_DATE_FORMAT_OPTIONS, ...DEFAULT_INTL_WEEKDAY_FORMAT_OPTIONS },
+  DateTimeWeekday: {
+    ...DEFAULT_INTL_DATE_FORMAT_OPTIONS,
+    ...DEFAULT_INTL_TIME_FORMAT_OPTIONS,
+    ...DEFAULT_INTL_WEEKDAY_FORMAT_OPTIONS,
+  },
+} satisfies Record<string, Intl.DateTimeFormatOptions>
+
+/**
+ * Formats a date into a string representation. If the date is within the date threshold, the date will be formatted as a relative date.
+ *
+ * @param date - The date to format.
+ * @param options - Optional formatting options.
+ * @param options.forceAbsolute - Whether the date will always be formatted as an absolute date.
+ * @param options.includeTime - Whether to include the time in the formatted date.
+ * @param options.includeWeekday - Whether to include the weekday in the formatted date.
+ * @param options.locale - The locale to use for formatting. Defaults to `nb-NO`.
+ * @param options.relativeDateThresholdDays - The number of days to use as a threshold for relative date formatting. Defaults to `3`.
+ * @returns The formatted date string.
+ */
+export const formatDate = (
+  date: Date,
+  options?: {
+    forceAbsolute?: boolean
+    includeTime?: boolean
+    includeWeekday?: boolean
+    locale?: Intl.LocalesArgument
+    relativeDateThresholdDays?: number
+  }
+): string => {
+  const daysDifference = differenceInDays(new Date(), date)
```

Comment:
If this is the only place you `date-fns` is used I'd rather replace it with a comparison of unix epochs—this seems to only cause cosmetic "issues", and we don't have to be careful regarding leap-years/timezones/other shenanigans here